apps/opik-frontend/src/components/pages/CompareExperimentsPage/CompareExperimentsPanel/OutputTab/ExperimentDataset.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useMemo } from "react";
+
+import { extractImageUrls } from "@/lib/images";
+import { DatasetItem } from "@/types/datasets";
+import { Button } from "@/components/ui/button";
+import { Braces } from "lucide-react";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuCustomCheckboxItem,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import uniq from "lodash/uniq";
+import ExperimentDatasetItems from "@/components/pages/CompareExperimentsPage/CompareExperimentsPanel/OutputTab/ExperimentDatasetItems";
+import useLocalStorageState from "use-local-storage-state";
+
+interface ExperimentDatasetProps {
+  data: DatasetItem["data"] | undefined;
+}
+
+const isASubsetB = (arrA: string[], arrB: string[]) => {
+  return arrA.every((a) => arrB.includes(a));
+};
+
+const IMAGES_KEY = "images";
+const SELECTED_DATA_SET_ITEM_KEYS =
+  "experiment-sidebar-selected-dataset-item-keys";
+
+const ExperimentDataset = ({ data }: ExperimentDatasetProps) => {
+  const [selectedKeys, setSelectedKeys] = useLocalStorageState<string[] | null>(
+    SELECTED_DATA_SET_ITEM_KEYS,
+    {
+      defaultValue: null,
+    },
+  );
+
+  const imagesUrls = useMemo(() => extractImageUrls(data), [data]);
+  const hasImages = imagesUrls.length > 0;
+
+  const dataKeys = useMemo(() => {
+    const keys = Object.keys(data || {});
+
+    if (hasImages && !keys.includes(IMAGES_KEY)) {
+      keys.push(IMAGES_KEY);
+    }
+
+    return uniq(keys);
+  }, [data, hasImages]);
+
+  const handleCheckChange = (key: string) => {
+    setSelectedKeys((selectedKeys) => {
+      if (selectedKeys?.includes(key)) {
+        return selectedKeys.filter((sk) => sk !== key);
+      }
+
+      return [...(selectedKeys || []), key];
+    });
+  };
+
+  useEffect(() => {
+    setSelectedKeys((sk) => {
+      if (sk === null || !isASubsetB(sk, dataKeys)) {
```

Comment:
I think to make it work you need to implement something like here - https://github.com/comet-ml/opik/blob/177ae1bea60ccd2644339fbfe12bf206da8cf795/apps/opik-frontend/src/components/pages/DatasetItemsPage/DatasetItemsPage.tsx#L142